app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -797,8 +797,8 @@ public static ReadingData parseData(int attempt, String tagId, byte[] data, Long
             //       getGlucose(new byte[]{data[(i * 6 + 125)], data[(i * 6 + 124)]});
 
             // If the data is decoded for some reason, we might have a wrong index.
-
-            if(i * 6 + 125 >= data.length) {
+            // The 6 is because we read up to 6 bytes.
```

Comment:
Are up to or exactly six bytes more read? It's not obvious to me where these extra bytes are read. Maybe, a constant should be used here and at the other place.

Reply:
A few lines later `data[(i * 6 + 125)]` and `data[(i * 6 + 124)]` are passed to a function. That's hard to understand. Maybe both `6` and `124` could be defined as a constant for more readability.

Reply:
I have replaced the code reading the data with constants. Also added a link to documentation.
I hope things are more readable now.